src/hooks/use-app-bridge/use-app-bridge.tsx

```diff
@@ -1,5 +1,6 @@
 import * as React from 'react';
-import {AppProviderContext} from '../../components';
+// eslint-disable-next-line shopify/strict-component-boundaries
```

Comment:
I was running into circular dependency issues. I couldn't find the culprit so I improved from the file directly and that worked 👌 Do you have any thoughts on this? @BPScott

Reply:
Circular dependencies and strict-component-boundary violations are usually a sign that you've got something that is a dependency of multiple things nested deeply inside part of the components folder and then things get funky when they start to be interdepent.

In the past I've found the solution is to pull up that common dependency to a higher level to avoid things importing from siblings. That removes the cicularity of dependencies between siblings as you end up with two siblings that both depend upon a 3rd file that's higher up the hierarchy instead of on each other.

---

I saw in https://github.com/Shopify/polaris-react/pull/1482 you added AppProviderContext, ThemeProviderContext and PolarisContext to src/components/index's exports. I think this is part of the root cause of this - you have something in hooks depending on something in components (namely this context), and something in components depending on this hook.

We don't want to expose these contexts as part of our public API - which adding them to src/components/index will do.

I think the solution to both of these problems (circular deps and accidentally making that public) is to move these global contexts that are consumed by both things in in /components/ and /hook/ out of components and into src/contexts/. At which point both this file, and components/AppProvider will `import {AppProviderContext} from 'ROOTPATHHERE/src/components/context';`.

Reply:
I gave moving `AppProvinder/ThemeProvider` context up into `src/contexts` a try however I still ran into circular dependencies (different ones). I'll create an issue to discuss re-architecting app provider and it's utilities. Since some of it's utilities will need to be moved out as well, e.g. `StickyManager`, `Link`, etc.